app/forms/pages/type_of_answer_form.rb

```diff
@@ -1,11 +1,28 @@
 class Pages::TypeOfAnswerForm < BaseForm
-  attr_accessor :answer_type, :form, :page
+  attr_accessor :answer_type, :draft_question
 
   validates :answer_type, presence: true, inclusion: { in: Page::ANSWER_TYPES }
 
-  def submit(session)
+  def submit
     return false if invalid?
 
-    session[:page] = { answer_type:, answer_settings: nil }
```

Comment:
this line is extremely destructive and can leave a session in a broken state, if the user is clicking back and changing previous answer types and then decides to go back to "Edit question" and saves theres nothing to catch that answer settings is nil